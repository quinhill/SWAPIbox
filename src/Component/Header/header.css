header {
  height: 200px;
  display: grid;
  grid-template-columns: 30% 40% 30%;
}

.logo-container{
  grid-column: 2 / 3;
  padding: 20px;
}

.logo {
  height: 160px;
}

.fav-button-container {
  padding: 20px;
}

.show-favorites {
  float: right;
  height: 50px;
  width: 100px;
  color: black;
  background-color: #FFC343;
  border: none;
}

.counter {
  background-color: #000000;
  color: #FFC343;
  margin: 5px;
  width: 20px;
  height: 20px;
}