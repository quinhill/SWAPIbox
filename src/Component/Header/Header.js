import React from 'react';
import './header.css';
import PropTypes from 'prop-types';

const Header = ({favorites, displayFavorites}) => {
  return (
    <header>
        <div className="logo-container">
        <img className="logo" alt="star wars logo" src="star-wars-logo.svg" />
      </div>
      <div className="fav-button-container">
        <button 
          className="show-favorites" 
          onClick={displayFavorites}
        >
          View Favorites
          <p className="counter">{favorites.length}</p>
        </button>
      </div>
    </header>
  )
}

Header.Proptypes = {
  favorites: PropTypes.arrayOf(PropTypes.object),
  displayFavorites: PropTypes.func
}

export default Header;