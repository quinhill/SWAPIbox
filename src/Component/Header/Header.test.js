import React from 'react';
import Header from './Header';
import { shallow } from 'enzyme';

describe('Header', () => {

  it('should match snapshot', () => {

    const mockFavorites = [{}, {}]

    const wrapper = shallow(<Header 
                            favorites={mockFavorites}
                            />)

    expect(wrapper).toMatchSnapshot()
  })
})