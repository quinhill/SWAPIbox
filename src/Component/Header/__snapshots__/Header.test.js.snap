// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="logo-container"
>
          <img
                    alt="star wars logo"
                    className="logo"
                    src="star-wars-logo.svg"
          />
</div>,
        <div
          className="fav-button-container"
>
          <button
                    className="show-favorites"
          >
                    View Favorites
                    <p
                              className="counter"
                    >
                              0
                    </p>
          </button>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="star wars logo"
            className="logo"
            src="star-wars-logo.svg"
/>,
          "className": "logo-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "star wars logo",
            "className": "logo",
            "src": "star-wars-logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="show-favorites"
>
            View Favorites
            <p
                        className="counter"
            >
                        0
            </p>
</button>,
          "className": "fav-button-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "View Favorites",
              <p
                className="counter"
>
                0
</p>,
            ],
            "className": "show-favorites",
          },
          "ref": null,
          "rendered": Array [
            "View Favorites",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "0",
                "className": "counter",
              },
              "ref": null,
              "rendered": "0",
              "type": "p",
            },
          ],
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="logo-container"
>
            <img
                        alt="star wars logo"
                        className="logo"
                        src="star-wars-logo.svg"
            />
</div>,
          <div
            className="fav-button-container"
>
            <button
                        className="show-favorites"
            >
                        View Favorites
                        <p
                                    className="counter"
                        >
                                    0
                        </p>
            </button>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="star wars logo"
              className="logo"
              src="star-wars-logo.svg"
/>,
            "className": "logo-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "star wars logo",
              "className": "logo",
              "src": "star-wars-logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="show-favorites"
>
              View Favorites
              <p
                            className="counter"
              >
                            0
              </p>
</button>,
            "className": "fav-button-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "View Favorites",
                <p
                  className="counter"
>
                  0
</p>,
              ],
              "className": "show-favorites",
            },
            "ref": null,
            "rendered": Array [
              "View Favorites",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "0",
                  "className": "counter",
                },
                "ref": null,
                "rendered": "0",
                "type": "p",
              },
            ],
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
